File: java/lib/src/main/java/org/sammancoaching/library/LibraryDemo.java
package org.sammancoaching.library;
public final class LibraryDemo {
    private LibraryDemo() {
    }
    public static void main(String[] args) {
        Member alice = new Member("MEMBER-001", "Alice");
        Member bob = new Member("MEMBER-002", "Bob");

        Book book = Book.available(
                "Refactoring",
                Isbn.of("978-0201485677"),
                "BOOK-001"
        );

        Book borrowedBook = book.borrow(alice);

        System.out.println("Borrowed by: " + borrowedBook.borrowedBy().orElseThrow().name());
        System.out.println("Due date: " + borrowedBook.dueDate());

        try {
            borrowedBook.borrow(bob);
        } catch (IllegalStateException exception) {
            System.out.println("Bob cannot borrow it: " + exception.getMessage());
        }
    }
}

File: csharp/Library/LibraryDemo.cs
namespace SammanCoaching.Library;
public static class LibraryDemo
{
    public static void Main(string[] args)
    {
        var alice = new Member("MEMBER-001", "Alice");
        var bob = new Member("MEMBER-002", "Bob");

        var book = Book.Available(
            "Refactoring",
            Isbn.Of("978-0201485677"),
            "BOOK-001");

        var borrowedBook = book.Borrow(alice);

        Console.WriteLine($"Borrowed by: {borrowedBook.BorrowedBy!.Name}");
        Console.WriteLine($"Due date: {borrowedBook.DueDate}");

        try
        {
            borrowedBook.Borrow(bob);
        }
        catch (InvalidOperationException exception)
        {
            Console.WriteLine($"Bob cannot borrow it: {exception.Message}");
        }
    }
}
